server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -799,6 +800,86 @@ public Consumer verifyAndLookupConsumerWithEntitlements(String consumerUuid) {
         return listByCriteria(crit, pageRequest);
     }
 
+    public int countConsumers(String ownerKey,
+        Set<String> typeLabels, List<String> skus, List<String> subscriptionIds, List<String> contracts) {
+
+        if (ownerKey == null || ownerKey.isEmpty()) {
+            throw new IllegalArgumentException("Owner key can't be null or empty");
+        }
+
+        Criteria crit = createCriteria(ownerKey, typeLabels, skus, subscriptionIds, contracts);
+        return ((Long) crit.uniqueResult()).intValue();
+    }
+
+    /**
+     *  JPQL of below criteria can look like this.
+     *  If all parameters arn't passed then only sub-parts of it are returned.
+     *
+     *   select count(distinct c.id) from Consumer c join c.owner o
+     *       join c.type ct
+     *       join c.entitlements e
+     *       join e.pool po
+     *       join po.product pr
+     *       join pr.attributes pa
+     *       join po.sourceSubscription ss
+     *   where o.key = :key
+     *   and ct.label in (...)
+     *   and pr.id in (...)
+     *   and pa.name = 'type'
+     *   and pa.value = 'MKT'
+     *   and ss.subscriptionId in (...)
+     *   and po.contractNumber in (...)
+     *
+     */
+    private Criteria createCriteria(String ownerKey, Set<String> typeLabels,
+        List<String> skus, List<String> subscriptionIds,
+        List<String> contracts) {
+
+        Criteria crit = super.createSecureCriteria("c");
+        crit.createAlias("c.owner", "o")
+            .add(Restrictions.eq("o.key", ownerKey));
+
+        if (hasSomeNotNullString(typeLabels)) {
```

Comment:
In case, when we have correct values plus null values everything is good, but in case when collections contain only null values then we are bad. Without removing nulls from collections, test named : 'blankCriteriaShouldCountAllExistingConsumers' won't pass.